source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -71,6 +71,55 @@ RoleBasedAccessControlFilterConfig::RoleBasedAccessControlFilterConfig(
       shadow_engine_(Filters::Common::RBAC::createShadowEngine(
           proto_config, context, validation_visitor, action_validation_visitor_)) {}
 
+std::string RoleBasedAccessControlFilterConfig::shadowEffectivePolicyIdField(
```

Comment:
the 4 functions all have the same pattern, wondering if we could do something (like templating) to avoid duplicating the code?